pkg/tenant/db/topics.go

```diff
@@ -0,0 +1,135 @@
+package db
+
+import (
+	"context"
+	"time"
+
+	"github.com/oklog/ulid/v2"
+	"github.com/vmihailenco/msgpack/v5"
+)
+
+const TopicNamespace = "topics"
+
+type Topic struct {
+	ProjectID ulid.ULID `msgpack:"project_id"`
+	ID        ulid.ULID `msgpack:"id"`
+	Name      string    `msgpack:"name"`
+	Created   time.Time `msgpack:"created"`
+	Modified  time.Time `msgpack:"modified"`
+}
+
+var _ Model = &Topic{}
+
+// Key is a 32 composite key combining the project ID and the topic ID.
+func (t *Topic) Key() (key []byte, err error) {
+	// Create a 32 byte array so that the first 16 bytes hold the project ID
+	// and the last 16 bytes hold the topic ID.
+	key = make([]byte, 32)
+
+	// Marshal the project ID to the first 16 bytes of the key.
+	if err = t.ProjectID.MarshalBinaryTo(key[0:16]); err != nil {
+		return nil, err
+	}
+
+	// Marshal the topic ID to the last 16 bytes of the key.
+	if err = t.ID.MarshalBinaryTo(key[16:]); err != nil {
+		return nil, err
+	}
+
+	return key, err
+}
+
+func (t *Topic) Namespace() string {
+	return TopicNamespace
+}
+
+func (t *Topic) MarshalValue() ([]byte, error) {
+	return msgpack.Marshal(t)
+}
+
+func (t *Topic) UnmarshalValue(data []byte) error {
+	return msgpack.Unmarshal(data, t)
+}
+
+// CreateTopic adds a new topic to the database.
+func CreateTopic(ctx context.Context, topic *Topic) (err error) {
+	if topic.ID.Compare(ulid.ULID{}) == 0 {
+		topic.ID = ulid.Make()
+	}
```

Comment:
We should probably create a story to add data validation here, e.g. in the case that there is no `ProjectID` we should return an error; topics should also have their topic names validated; they can only contain alphanumerics and _ and cannot start with a number. 

We can create a `Validate()` method on the model that we can call before create and and update to ensure that our data structure is valid.